services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/ProtectedPathSpec.groovy

```diff
@@ -909,6 +909,80 @@ class ProtectedPathSpec extends BaseSpecification {
         database.resetCosts()
     }
 
+    def "Able to synchronize rules for a flow with protected path"() {
+        given: "Two active not neighboring switches"
+        def switches = topology.getActiveSwitches()
+        def allLinks = northbound.getAllLinks()
+        def (Switch srcSwitch, Switch dstSwitch) = [switches, switches].combinations()
+                .findAll { src, dst -> src != dst }.find { Switch src, Switch dst ->
+            allLinks.every { link -> !(link.source.switchId == src.dpId && link.destination.switchId == dst.dpId) }
+        } ?: assumeTrue("No suiting switches found", false)
+
+        and: "Create a flow with protected path"
+        def flow = flowHelper.randomFlow(srcSwitch, dstSwitch)
+        flow.allocateProtectedPath = true
+        flowHelper.addFlow(flow)
+
+        def mainFlowPath = northbound.getFlowPath(flow.id).forwardPath
+        def protectedFlowPath = northbound.getFlowPath(flow.id).protectedPath.forwardPath
+        def commonNodeIds = mainFlowPath*.switchId.intersect(protectedFlowPath*.switchId)
+        Wrappers.wait(WAIT_OFFSET) {
+            verifyRulesOnCommonNodes(commonNodeIds, srcSwitch.dpId, dstSwitch.dpId,
+                    mainFlowPath, protectedFlowPath)
+        }
+
+        def uniqueNodes = protectedFlowPath.findAll { !commonNodeIds.contains(it.switchId) } + mainFlowPath.findAll {
+            !commonNodeIds.contains(it.switchId)
+        }
+        verifyRulesOnUniqueSwitches(uniqueNodes)
+
+        and: "Reproduce situation when switches have missing rules by deleting flow rules from them"
+        commonNodeIds.each { northbound.deleteSwitchRules(it, DeleteRulesAction.IGNORE_DEFAULTS) }
+        uniqueNodes.each { northbound.deleteSwitchRules(it.switchId, DeleteRulesAction.IGNORE_DEFAULTS) }
+
+        commonNodeIds.each { switchId ->
+            if (switchId == srcSwitch.dpId || switchId == dstSwitch.dpId) {
+                assert northbound.validateSwitchRules(switchId).missingRules.size() == 3
+            } else {
+                assert northbound.validateSwitchRules(switchId).missingRules.size() == 4
+            }
+        }
+
+        uniqueNodes.each { assert northbound.validateSwitchRules(it.switchId).missingRules.size() == 2 }
+
+        when: "Synchronize rules on switches"
+        commonNodeIds.each { northbound.synchronizeSwitchRules(it) }
+        uniqueNodes.each { northbound.synchronizeSwitchRules(it.switchId) }
```

Comment:
Note that you are not verifying the correctness of responses to `synchronizeSwitchRules`

Reply:
I guess we have separate test for this purpose.
here I just check result of synchronization

should I check response to `synchronizeSwitchRules` ?

Reply:
added